classes/Kohana/Model/Promotion.php

```diff
@@ -93,13 +93,18 @@ public function update_brand_purchase_items($applies, & $items)
 	{
 		$promo_item = Jam::build('purchase_item_promotion', array('reference' => $this));
 
-		$items = array_filter($items, function($item) use ($promo_item) {
-			return ! $item->is_same($promo_item);
-		});
+		$existing_promotion = false;
```

Comment:
Better var name should be `has_existing_promotion` since its a boolean, not an actual promotion.